src/Components/test/testassets/BasicTestApp/Program.cs

```diff
@@ -42,6 +44,16 @@ public static async Task Main(string[] args)
                     policy.RequireAssertion(ctx => ctx.User.Identity.Name?.StartsWith("B") ?? false));
             });
 
+            // Replace the default logger with a custom one that wraps it
+            var originalLoggerDescriptor = builder.Services.Single(d => d.ServiceType == typeof(ILoggerFactory));
+            builder.Services.AddSingleton<ILoggerFactory>(services =>
+            {
+                var originalLogger = (ILoggerFactory)Activator.CreateInstance(
+                    originalLoggerDescriptor.ImplementationType,
+                    new object[] { services });
+                return new PrependMessageLoggerFactory("Custom logger", originalLogger);
+            });
+
```

Comment:
Do we need APIs to configure this in the WebAssemblyHostbuilder? It seems cumbersome to do at the moment and I can already think of people wanting to configure things like application insights and other telemetry tools here.

Reply:
Discussed offline. Will file an issue to consider this as a future enhancement.

The right thing to do in the longer term will be to replicate all the logic from `LoggerFactory` and `Logger` in https://github.com/dotnet/extensions/blob/master/src. The reason we're not using that package's implementation directly is because of size concerns.

Reply:
Filed https://github.com/dotnet/aspnetcore/issues/19737